### PR TITLE
tests for Category

### PR DESCRIPTION
**Pull request by OlegKandalov**

### CODE REVIEW

File: src/test/java/com/exadel/discount/platform/web/CategoryControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    private final CategoryService categoryService;
+    protected final WebApplicationContext context;
+    private static final UUID deletedCategory = UUID.fromString("7c90994b-06f4-4e42-aafb-c9b07eeb9815");
+    private static final UUID actualCategory = UUID.fromString("a55b8b0c-cd2b-4896-a074-bafb558266a9");
+    private static final UUID actualSubCategory = UUID.fromString("35f97267-404e-4d95-98f6-9a399bc3a9de");
+
+    protected MockMvc mockMvc;
+    protected static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
+            .setSerializationInclusion(JsonInclude.Include.NON_NULL);
```
Comment (ohalushchak-exadel) [inline src/test/java/com/exadel/discount/platform/web/CategoryControllerTest.java:40]: private

File: src/test/java/com/exadel/discount/platform/web/CategoryControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+import java.util.List;
+import java.util.UUID;
+
+@SpringBootTest
+@WebAppConfiguration
+public class CategoryControllerTest {
+
+    private final CategoryService categoryService;
+    protected final WebApplicationContext context;
```
Comment (ohalushchak-exadel) [inline src/test/java/com/exadel/discount/platform/web/CategoryControllerTest.java:33]: private

File: src/test/java/com/exadel/discount/platform/web/CategoryControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        byte[] contentAsByte = mockMvc.perform(MockMvcRequestBuilders.get("/category")
+                .param("isDeleted", "true"))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andReturn().getResponse().getContentAsByteArray();
+
+        List<CategoryResponseDto> categories = OBJECT_MAPPER.readValue(contentAsByte
+                , new TypeReference<List<CategoryResponseDto>>() {
+                });
+
+        Assertions.assertEquals(2, categories.size());
```
Comment (ohalushchak-exadel) [inline src/test/java/com/exadel/discount/platform/web/CategoryControllerTest.java:67]: check the actual content of the list (that all fields are present/set)

File: src/test/java/com/exadel/discount/platform/web/CategoryControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+        CategoryDto category = new CategoryDto();
+        category.setName("RestTest");
+
+        MockHttpServletRequestBuilder requestBuilder = MockMvcRequestBuilders.post("/category")
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(OBJECT_MAPPER.writeValueAsString(category));
+
+        mockMvc.perform(requestBuilder)
+                .andExpect(MockMvcResultMatchers.status().isCreated())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.name").value("RestTest"))
```
Comment (ohalushchak-exadel) [inline src/test/java/com/exadel/discount/platform/web/CategoryControllerTest.java:92]: check that id is returned as well

File: src/test/java/com/exadel/discount/platform/web/CategoryControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+                });
+
+        Assertions.assertEquals(2, categories.size());
+    }
+
+    @Test
+    @WithMockUser(username = "testUser", authorities = "USER")
+    @Sql(value = {"/insertDeletedCategoryTest.sql"}, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(value = {"/deleteCategoryTest.sql"}, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    public void shouldGetAllDeletedCategoryForUser() throws Exception {
```
Comment (ohalushchak-exadel) [inline src/test/java/com/exadel/discount/platform/web/CategoryControllerTest.java:74]: shouldNot:)

File: src/test/resources/deleteCategoryTest.sql [hunk 6]
```diff
@@ -0,0 +1,6 @@
+DELETE FROM category WHERE c_id = '7c90994b-06f4-4e42-aafb-c9b07eeb9815';
```
Comment (ohalushchak-exadel) [inline src/test/resources/deleteCategoryTest.sql:1]: just delete everything from both tables

Comment (OlegKandalov) [reply src/test/resources/deleteCategoryTest.sql:1]: done

**Review by nkharitoniuk — APPROVED** [review]

**Pull request merged**